data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java

```diff
@@ -61,18 +71,38 @@ private OTelTraceSourceConfig(final int requestTimeoutInMillis,
         this.sslKeyCertChainFile = sslKeyCertChainFile;
         this.sslKeyFile = sslKeyFile;
         this.acmCertificateArn = acmCertificateArn;
+        this.acmPrivateKeyPassword = acmPrivateKeyPassword;
+        this.awsRegion = awsRegion;
         this.threadCount = threadCount;
         this.maxConnectionCount = maxConnectionCount;
-        if (useAcmCertForSSL && (acmCertificateArn == null || acmCertificateArn.isEmpty())) {
-            throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", USE_ACM_CERT_FOR_SSL, ACM_CERT_ARN));
-        } else {
-            if (ssl && (sslKeyCertChainFile == null || sslKeyCertChainFile.isEmpty())) {
-                throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", SSL, SSL_KEY_CERT_FILE));
-            }
-            if (ssl && (sslKeyFile == null || sslKeyFile.isEmpty())) {
-                throw new IllegalArgumentException(String.format("%s is enabled, %s can not be empty or null", SSL, SSL_KEY_CERT_FILE));
+        boolean certAndKeyFileInS3 = false;
+        if (useAcmCertForSSL) {
+            validateSSLArgument(String.format("%s is enabled", USE_ACM_CERT_FOR_SSL), acmCertificateArn, ACM_CERT_ARN);
+            validateSSLArgument(String.format("%s is enabled", USE_ACM_CERT_FOR_SSL), awsRegion, AWS_REGION);
```

Comment:
Now that config values are being passed into the CertProvider implementation constructors, we can use `Objects.requireNonNull()` checks at the constructor level instead of manually verifying here. (Sorry this just occurred to me).

Reply:
Will add this.